src/clients/api/queries/getHypotheticalAccountLiquidity.ts

```diff
@@ -0,0 +1,37 @@
+import BigNumber from 'bignumber.js';
+import { Contract } from 'web3-eth-contract';
+import { Asset } from 'types';
+
+export interface IGetHypotheticalAccountLiquidityInput {
+  comptrollerContract: Contract; // @TODO: use contract type (through Typechain?)
+  account: string | undefined | null;
+  asset: Asset;
+  balanceOf: string;
+  borrowAmount?: number | BigNumber;
+}
+
+export type GetHypotheticalAccountLiquidityOutput = number;
+
+const getHypotheticalAccountLiquidity = async ({
+  comptrollerContract,
+  account,
+  asset,
+  balanceOf,
+  borrowAmount = 0,
+}: IGetHypotheticalAccountLiquidityInput) => {
+  const a = { ...asset };
+  const response = await comptrollerContract.methods
+    .getHypotheticalAccountLiquidity(
+      account?.toLowerCase(),
+      asset.vtokenAddress,
+      balanceOf,
+      borrowAmount,
+    )
+    .call();
+  if (response) {
+    a.hypotheticalLiquidity = response;
+  }
+  return a;
```

Comment:
We could do that but there are so many loops in the hook that I decided to follow this original pattern to avoid another loop. I was able to refactor some out but not all

Reply:
I was able to refactor this.